cli/utils.go

```diff
@@ -0,0 +1,209 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/spf13/viper"
+
+	"github.com/prometheus/alertmanager/types"
+)
+
+const (
+	RESOLVE_EXACT = iota
+	RESOLVE_FUZZY = iota
+)
+
+func GetAlertmanagerURL() (*url.URL, error) {
+	u, err := url.ParseRequestURI(viper.GetString("alertmanager.url"))
+	if err != nil {
+		return nil, errors.New("Invalid alertmanager url")
+	}
+	return u, nil
+}
+
+// Parse a list of labels (cli arguments)
+func parseMatchers(labels []string, resolve int) (types.Matchers, error) {
```

Comment:
Quick question.
I implemented a feature where if a user doing a query (silence or alert) does something like the following:
    amtool alert query Host

With no matcher name specified then I attempt to match the alertname using a prefix match. Is this something we want to support? Moving to promql should do everything I want, but i'm curious if I should spend the time to re-implement this functionality?